### ISSUE TITLE
[Enhancement] Add TextCaseTypeLocal.FirstUpperRestLower to TextCaseConverter

### ISSUE THREAD

**Issue by JoacimWall** [xamarin/XamarinCommunityToolkit#762]
## Summary

Wi use the converters in toolkit but we are missing one feature in the TextCaseConverter. We using many string in Resource files and somtime this need to me Lower,Upper and somtime First upper the rest lower. 

if we could add this feture to the TextCaseConverter

Regards Joacim Wall 

 

## API Changes

```csharp
public enum TextCaseType
	{
		None,
		Upper,
		Lower,
--New	FirstUpperRestLower, 
	}
```

add one line of code to the 
```csharp
object Convert(string value, object parameter)
	=> GetParameter(parameter) switch
	{
		TextCaseTypeLocal.Lower => value?.ToLowerInvariant(),
		TextCaseTypeLocal.Upper => value?.ToUpperInvariant(),
--New.     TextCaseTypeLocal.FirstUpperRestLower => char.ToUpper((char)(value?.First())) + value.Substring(1).ToLower(),
		_ => value
	};
```

## Intended Use Case
We using many string in Resource files and somtime this need to me Lower,Upper and somtime First upper the rest lower. 

## Who Will Do The Work?

- [X] I am willing to take this on myself
- [ ] Just putting this out there for someone to pick up

**Comment by AndreiMisiukevich** [comment]
@JoacimWall feature approve)

**Comment by JoacimWall** [comment]
Hi 
Shoud i now make a PR request for brach Develop?

**Comment by AndreiMisiukevich** [comment]
> Hi
> Shoud i now make a PR request for branch Develop?

Yes please

**Comment by JoacimWall** [comment]
has created a PR #763

**Comment by pictos** [comment]
closed by #763
